### PR TITLE
CPM-999: Apply complex filter with match

### PR DESCRIPTION
**Pull request by antoinearrive**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Category/back/Infrastructure/Symfony/Resources/config/queries.yml [hunk 1]
```diff
@@ -6,3 +6,8 @@ services:
         class: Akeneo\Pim\Enrichment\Category\Infrastructure\Query\SqlGetExistingCategories
         arguments:
             - '@database_connection'
+            -
```
Comment (Beninho) [inline src/Akeneo/Pim/Enrichment/Category/back/Infrastructure/Symfony/Resources/config/queries.yml:9]: empty argument ?

File: components/identifier-generator/back/src/Application/Match/Condition/MatchCategoryHandler.php [hunk 2]
```diff
@@ -14,6 +17,11 @@
  */
 class MatchCategoryHandler implements MatchConditionHandler
```
Comment (mmetayer) [inline components/identifier-generator/back/src/Application/Match/Condition/MatchCategoryHandler.php:18]: final

**Review by mmetayer — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Category/back/Infrastructure/Query/SqlCategoriesHaveAtLeastOneChild.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Category\Infrastructure\Query;
```
Comment (amigaire) [inline src/Akeneo/Pim/Enrichment/Category/back/Infrastructure/Query/SqlCategoriesHaveAtLeastOneChild.php:5]: Could you put the query implementation in Akeneo\Category\Infrastructure\Storage\Sql plz?

File: src/Akeneo/Pim/Enrichment/Category/back/API/Query/CategoriesHaveAtLeastOneChild.php [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Category\API\Query;
```
Comment (amigaire) [inline src/Akeneo/Pim/Enrichment/Category/back/API/Query/CategoriesHaveAtLeastOneChild.php:5]: Could you put it in Akeneo\Category\ServiceApi\Query plz?

**Comment by amigaire** [comment]
GTM after move of the query in new Category Bounded Context

**Pull request merged**